internal/tasks/account_management_test.go

```diff
@@ -0,0 +1,172 @@
+		`,
+		s.Clock.Now().Add(1*time.Hour).Unix())
+
+	// test if errors are propagated
+	s.AMD.ConfigPath = "./fixtures/account_management_error.json"
+	s.Clock.StepBy(2 * time.Hour)
+	expectError(t, fmt.Sprintf("could not configure managed account %q: %s", "", processor.ErrAccountNameEmpty.Error()), job.ProcessOne(s.Ctx))
+	tr.DBChanges().AssertEmpty()
+
+	// and delete the account again
+	s.AMD.ConfigPath = "./fixtures/account_management_empty.json"
+	s.Clock.StepBy(2 * time.Hour)
+	expectSuccess(t, job.ProcessOne(s.Ctx))
+	expectError(t, sql.ErrNoRows.Error(), job.ProcessOne(s.Ctx))
+	tr.DBChanges().AssertEqualf(`
+		DELETE FROM accounts WHERE name = 'abcde';
+	`)
+}
+
+func TestAccountManagementWithComplexDeletion(t *testing.T) {
+	j, s := setup(t)
+	job := j.EnforceManagedAccountsJob(s.Registry)
+
+	tr, tr0 := easypg.NewTracker(t, s.DB.DbMap.Db)
+	tr0.Ignore()
+
+	// create a managed account named "abcde"
+	s.AMD.ConfigPath = "./fixtures/account_management_basic.json"
+	expectSuccess(t, job.ProcessOne(s.Ctx))
+	expectError(t, sql.ErrNoRows.Error(), job.ProcessOne(s.Ctx))
+	tr.DBChanges().Ignore()
+
+	// give quota to its auth tenant
+	mustDo(t, s.DB.Insert(&models.Quotas{
+		AuthTenantID:  "12345",
+		ManifestCount: 100,
+	}))
+
+	// upload an image into that account
+	repo := models.Repository{
+		AccountName: "abcde",
+		Name:        "foo",
+	}
+	image := test.GenerateImage(
+		test.GenerateExampleLayer(1),
+		test.GenerateExampleLayer(2),
+	)
+	image.MustUpload(t, s, repo, "latest")
+	tr.DBChanges().Ignore()
+
+	// try to delete the managed account: first attempt will set in_maintenance and delete the image, but nothing more
+	s.AMD.ConfigPath = "./fixtures/account_management_empty.json"
+	s.Clock.StepBy(2 * time.Hour)
+	expectSuccess(t, job.ProcessOne(s.Ctx))
+	tr.DBChanges().AssertEqualf(`
+		UPDATE accounts SET in_maintenance = TRUE, next_enforcement_at = %[1]d WHERE name = 'abcde';
+		DELETE FROM manifest_blob_refs WHERE repo_id = 2 AND digest = '%[2]s' AND blob_id = 1;
+		DELETE FROM manifest_blob_refs WHERE repo_id = 2 AND digest = '%[2]s' AND blob_id = 2;
+		DELETE FROM manifest_blob_refs WHERE repo_id = 2 AND digest = '%[2]s' AND blob_id = 3;
+		DELETE FROM manifest_contents WHERE repo_id = 2 AND digest = '%[2]s';
+		DELETE FROM manifests WHERE repo_id = 2 AND digest = '%[2]s';
+		DELETE FROM tags WHERE repo_id = 2 AND name = 'latest';
+		DELETE FROM trivy_security_info WHERE repo_id = 2 AND digest = '%[2]s';
+		`,
```

Comment:
gofmt sadly doesn't understand this

Reply:
Based on the indentation in existing instances of AssertEqualf, I have made a guess regarding what you mean by "this". Please take a look.